### PR TITLE
MediaViewer: iterate on design

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Media Viewer: design iteration.
Show the caption and change the design according to https://www.figma.com/design/Ni6Ii8YKtmXCKYNE90cC67/Timeline-(new)?node-id=1732-79703

image

## Tests

- There should be no behavior change, only UI change.

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_4_en.png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_4_en.png:1]: Note: there is an overlap here, but this is a preview issue. In prod `bottomPaddingInPixels` will be updated and the rendering will be OK.

File: libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/viewer/MediaViewerView.kt [hunk 1]
```diff
@@ -97,21 +110,37 @@ fun MediaViewerView(
...
+                    eventSink = state.eventSink
+                )
+                MediaViewerBottomBar(
+                    modifier = Modifier.align(Alignment.BottomCenter),
+                    actionsEnabled = state.downloadedMedia is AsyncData.Success,
+                    canDownload = state.canDownload,
+                    canShare = state.canShare,
+                    mimeType = state.mediaInfo.mimeType,
+                    caption = state.mediaInfo.caption,
+                    onHeightChanged = { bottomPaddingInPixels = it },
```
Comment (ganfra) [inline libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/viewer/MediaViewerView.kt:132]: why do we need this padding computation?

File: tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_3_en.png
Comment (ganfra) [inline tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_3_en.png:1]: This look weird to have the title centered and so big, have you raised the point it's not really "androidish"?

**Review by ganfra — COMMENTED** [review]
Some remarks/questions

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesFlowNode.kt [hunk 2]
```diff
@@ -337,7 +337,9 @@ class MessagesFlowNode @AssistedInject constructor(
                         caption = event.content.caption,
                         mimeType = event.content.mimeType,
                         formattedFileSize = event.content.formattedFileSize,
-                        fileExtension = event.content.fileExtension
+                        fileExtension = event.content.fileExtension,
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesFlowNode.kt:340]: dbb062f230db3499c8f70e6a90b45559e2de1815

File: libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/viewer/MediaViewerView.kt (hunk 1 above)
Comment (bmarty) [reply libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/viewer/MediaViewerView.kt:132]: Yes

File: tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_3_en.png
Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/libraries.mediaviewer.impl.viewer_MediaViewerView_3_en.png:1]: I raised the point, design team (@americanrefugee ) is not against it. My idea is to implement what has been designed first.

**Review by ganfra — APPROVED** [review]
Thanks!

**Pull request merged**